internal/consensus/metrics.gen.go

```diff
@@ -192,39 +192,48 @@ func PrometheusMetrics(namespace string, labelsAndValues ...string) *Metrics {
 		}, append(labels, "vote_type")).With(labelsAndValues...),
+		ProposalTimestampDifference: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: MetricsSubsystem,
+			Name:      "proposal_timestamp_difference",
+			Help:      "Difference between the timestamp in the proposal message and the local time of the validator at the time it received the message.",
+
+			Buckets: []float64{-10, -.5, -.025, 0, .1, .5, 1, 1.5, 2, 10},
+		}, append(labels, "is_timely")).With(labelsAndValues...),
 	}
 }
 
 func NopMetrics() *Metrics {
 	return &Metrics{
-		Height:                    discard.NewGauge(),
-		ValidatorLastSignedHeight: discard.NewGauge(),
-		Rounds:                    discard.NewGauge(),
-		RoundDurationSeconds:      discard.NewHistogram(),
-		Validators:                discard.NewGauge(),
-		ValidatorsPower:           discard.NewGauge(),
-		ValidatorPower:            discard.NewGauge(),
-		ValidatorMissedBlocks:     discard.NewGauge(),
-		MissingValidators:         discard.NewGauge(),
-		MissingValidatorsPower:    discard.NewGauge(),
-		ByzantineValidators:       discard.NewGauge(),
-		ByzantineValidatorsPower:  discard.NewGauge(),
-		BlockIntervalSeconds:      discard.NewHistogram(),
-		NumTxs:                    discard.NewGauge(),
-		BlockSizeBytes:            discard.NewGauge(),
-		TotalTxs:                  discard.NewGauge(),
-		CommittedHeight:           discard.NewGauge(),
-		BlockParts:                discard.NewCounter(),
-		DuplicateBlockPart:        discard.NewCounter(),
-		DuplicateVote:             discard.NewCounter(),
-		StepDurationSeconds:       discard.NewHistogram(),
-		BlockGossipPartsReceived:  discard.NewCounter(),
-		QuorumPrevoteDelay:        discard.NewGauge(),
-		FullPrevoteDelay:          discard.NewGauge(),
-		VoteExtensionReceiveCount: discard.NewCounter(),
-		ProposalReceiveCount:      discard.NewCounter(),
-		ProposalCreateCount:       discard.NewCounter(),
-		RoundVotingPowerPercent:   discard.NewGauge(),
-		LateVotes:                 discard.NewCounter(),
+		Height:                      discard.NewGauge(),
+		ValidatorLastSignedHeight:   discard.NewGauge(),
+		Rounds:                      discard.NewGauge(),
+		RoundDurationSeconds:        discard.NewHistogram(),
+		Validators:                  discard.NewGauge(),
+		ValidatorsPower:             discard.NewGauge(),
+		ValidatorPower:              discard.NewGauge(),
+		ValidatorMissedBlocks:       discard.NewGauge(),
+		MissingValidators:           discard.NewGauge(),
+		MissingValidatorsPower:      discard.NewGauge(),
+		ByzantineValidators:         discard.NewGauge(),
+		ByzantineValidatorsPower:    discard.NewGauge(),
+		BlockIntervalSeconds:        discard.NewHistogram(),
+		NumTxs:                      discard.NewGauge(),
+		BlockSizeBytes:              discard.NewGauge(),
```

Comment:
Is this supposed to be a gauge or histogram https://github.com/tendermint/tendermint/pull/7709/files#diff-8ba8e127ec86116d1c551254adca1654f36b94251c593748e05d91dcceb1f4beR267

Reply:
Good point.

Originally it was a Gauge: https://github.com/tendermint/tendermint/commit/489d9b9184f06d1c7e38f5895d0b0e42a9839aa3

This is the version on main.

Then it was changed to an histogram: https://github.com/tendermint/tendermint/commit/2f6141645bdc92a7f16653ce544a5ba1f4c907d5

This version is on master, but not on main.

Reply:
In any case, in `metrics.go` it is a Gauge (line 58).

Reply:
tendermint/tendermint#6549 (from 2f6141645 found by @cason above) is the commit that changed from Gauge to Histogram. This happened in the `v0.35.x`/`v0.36.x`/`master` timeline, but not on `v0.34.x`/`v0.37.x`/`v0.38.x`/`main`.

We can decide to also forward-port tendermint/tendermint#6549 to `main`, but that would be independent from the PBTS effort.

Reply:
Actually, I've just read tendermint/tendermint#5752, which is the issue tendermint/tendermint#6549 was addressing, and it may make sense to forward-port it to main, so I copied tendermint/tendermint#5752 into our repo: #2104